HW2/P2/parallel_vector.pyx

```diff
@@ -105,11 +110,16 @@ cpdef move_data_medium_grained(np.int32_t[:] counts,
    # to parallelize data movement.  Be sure to avoid deadlock, as well as
    # double-locking.
    ##########
-   with nogil:
-       for r in range(repeat):
-           for idx in range(src.shape[0]):
-               if counts[src[idx]] > 0:
-                   counts[dest[idx]] += 1
-                   counts[src[idx]] -= 1
+   for r in range(repeat):
+       for idx in prange(src.shape[0], nogil=True, num_threads=4):
+           if counts[src[idx]] > 0:
```

Comment:
you should acquire the lock before accessing counts! since counts could be changing!